overlord/devicestate/devicestate_install_mode_test.go

```diff
@@ -74,47 +100,87 @@ func (s *deviceMgrInstallModeSuite) makeMockInstalledPcGadget(c *C) {
 	snaptest.MockSnapWithFiles(c, "name: pc\ntype: gadget", si, [][]string{
 		{"meta/gadget.yaml", gadgetYaml},
 	})
-	s.makeModelAssertionInState(c, "canonical", "pc-model", map[string]interface{}{
+
+	si = &snap.SideInfo{
+		RealName: "core20",
+		Revision: snap.R(2),
+		SnapID:   core20SnapID,
+	}
+	snapstate.Set(s.state, "core20", &snapstate.SnapState{
+		SnapType: "base",
+		Sequence: []*snap.SideInfo{si},
+		Current:  si.Revision,
+		Active:   true,
+	})
+	snaptest.MockSnapWithFiles(c, "name: core20\ntype: base", si, nil)
+
+	s.mockModel = s.makeModelAssertionInState(c, "my-brand", "my-model", map[string]interface{}{
```

Comment:
can't we return the model from this helper instead of the using the field?

Reply:
or the model can be retrieved with the s.mgr.Model method

Reply:
Will do this is a small followup